src/common/QskRgbValue.h

```diff
@@ -235,4 +235,335 @@ namespace QskRgb
+        template< QRgb V >
+        constexpr QRgb parse_hex_literal_h()
+        {
+            return V;
+        }
+
+        template< QRgb V, char C, char... Cs >
+        constexpr QRgb parse_hex_literal_h()
+        {
+            constexpr auto is_0_9 = C >= '0' && C <= '9';
+            constexpr auto is_a_f = C >= 'a' && C <= 'f';
+            constexpr auto is_A_F = C >= 'A' && C <= 'F';
+
+            static_assert( is_0_9 || is_a_f || is_A_F, "invalid hex character" );
+
+            if constexpr ( is_0_9 )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - '0' ), Cs... >();
+            }
+            if constexpr ( is_a_f )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - 'a' + 10 ), Cs... >();
+            }
+            if constexpr ( is_A_F )
+            {
+                return parse_hex_literal_h< ( V << 4 ) | ( C - 'A' + 10 ), Cs... >();
+            }
+        }
+
+        template< char C, char... Cs >
+        constexpr QRgb parse_hex_literal_x()
+        {
+            static_assert( C == 'x', "invalid hex prefix character" );
+            return parse_hex_literal_h< 0, Cs... >();
+        }
+
+        template< char C, char... Cs >
+        constexpr QRgb parse_hex_literal_0()
+        {
+            static_assert( C == '0', "invalid hex prefix character" );
+            return parse_hex_literal_x< Cs... >();
+        }
+
+        namespace qrgb
+        {
+            // converts a hex string from '#RRGGBB[AA]' to '0xAARRGGBB' integer
+            QSK_EXPORT constexpr QRgb operator""_rgba(
+                const char* const str, const size_t len ) noexcept
+            {
+                constexpr auto max = 9;
+                const auto argb = static_cast< int >( fromHexString( str, len ) );
+                const auto r = ( argb >> ( len == max ? 24 : 16 ) ) & 0xFF;
+                const auto g = ( argb >> ( len == max ? 16 : 8 ) ) & 0xFF;
+                const auto b = ( argb >> ( len == max ? 8 : 0 ) ) & 0xFF;
+                const auto a = ( len == max ? argb & 0xFF : 0xFF );
+                return qRgba( r, g, b, a );
+            }
+
+            // converts a hex string from '#[AA]RRGGBB' to '0xAARRGGBB' integer
+            QSK_EXPORT constexpr QRgb operator""_argb(
+                const char* const str, const size_t len ) noexcept
+            {
+                constexpr auto max = 9;
+                const auto argb = static_cast< int >( fromHexString( str, len ) );
+                const auto r = ( argb >> 16 ) & 0xFF;
+                const auto g = ( argb >> 8 ) & 0xFF;
+                const auto b = ( argb >> 0 ) & 0xFF;
+                const auto a = ( len == max ? argb >> 24 : 0xFF ) & 0xFF;
+                return qRgba( r, g, b, a );
+            }
+
+            // converts a hex literal from '0xRRGGBB[AA]' to '0xAARRGGBB' integer
+            template< char... Cs >
+            constexpr QRgb operator""_rgba()
+            {
```

Comment:
@uwerat I had to resort to this `numeric literal operator template` since AFAIK there are no 32 bit literal operators (see: https://en.cppreference.com/w/cpp/language/user_literal). Thereby a user can type `0x112233_argb` or `0x112233_argb` and we can detect and default the missing alpha channel.